types/src/transaction/script.rs

```diff
@@ -50,3 +51,57 @@ impl fmt::Debug for Script {
             .finish()
     }
 }
+
+/// How to call a particular Move script (aka. an "ABI").
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Serialize, Deserialize)]
+pub struct ScriptABI {
+    /// The public name of the script.
+    name: String,
+    /// Some text comment.
+    doc: String,
+    /// The `code` value to set in the `Script` object.
+    code: Vec<u8>,
+    /// The names of the type arguments.
+    ty_args_names: Vec<String>,
+    /// The names and expected types of the regular arguments.
+    /// Note: by construction, these types never depend on the type arguments.
+    args_names_and_types: Vec<(String, TypeTag)>,
```

Comment:
is there an type-erasure happening here? to construct actual arguments we need to know the actual sub-type of TypeTag, right? 

plus, having this generic cotnainer inmplementation seems an overkill, I would settle to just have macro constrctut an struct and we can change the ScriptAPIView into an enum that reference these script struct manually.

Reply:
Let me add some example(s) tomorrow to illustrate what the generated code is going to look like on the client side.